### PR TITLE
First commits

### PR DESCRIPTION
**Pull request by jsf9k**
## 🗣 Description ##

This pull request implements the Python code for an AWS Lambda function that will disable users' inactive console and programmatic access.

See also:
- cisagov/provisionaccount-role-tf-module#45
- cisagov/disable-inactive-iam-users-tf-module#1
- cisagov/cool-accounts#147

## 💭 Motivation and context ##

We have a POAM about disabling inactive users in IAM.  Partially resolves cisagov/cool-system-internal#143.

## 🧪 Testing ##

I applied this module to our COOL Users account without incident.

## ✅ Pre-approval checklist ##

- [x] This PR has an informative and human-readable title.
- [x] Changes are limited to a single goal - *eschew scope creep!*
- [x] All relevant type-of-change labels have been added.
- [x] I have read the [CONTRIBUTING](../blob/develop/CONTRIBUTING.md) document.
- [x] These code changes follow [cisagov code standards](https://github.com/cisagov/development-guide).
- [x] All relevant repo and/or project documentation has been updated to reflect the changes in this PR.
- [x] All new and existing tests pass.

## ✅ Pre-merge checklist ##

- [x] Finalize version.

## ✅ Post-merge checklist ##

- [x] Create a release.

### CODE REVIEW

File: src/lambda_handler.py [hunk 1]
```diff
@@ -36,50 +40,159 @@ def task_default(event):
...
-    if denominator is None or numerator is None:
-        error_msg = "Request must include both a numerator and a denominator."
-        logging.error(error_msg)
-        failed_task(result, error_msg)
+    # Check that expiration_days can be interpreted as a strictly positive
+    # integer.
+    if "expiration_days" not in event:
+        errors.append('Missing required key "expiration_days" in event.')
+    elif not event["expiration_days"]:
+        errors.append('"account_ids" must be non-null.')
```
Comment (dav3r) [inline src/lambda_handler.py]: Looks like you copy/pasted this from elsewhere, but didn't fully update it.  Is this what you meant?
```suggestion
    elif not event["expiration_days"]:
        errors.append('"expiration_days" must be non-null.')
```

File: src/lambda_handler.py [hunk 2]
```diff
@@ -36,50 +40,159 @@ def task_default(event):
...
-            logging.error(error_msg)
-            logging.error(variable_error_msg, numerator, denominator)
-            failed_task(result, error_msg)
-        except ZeroDivisionError:
-            error_msg = "The denominator cannot be zero."
-            logging.error(error_msg)
-            logging.error(variable_error_msg, numerator, denominator)
-            failed_task(result, error_msg)
-
+            errors.append('"account_ids" must be an integer.')
```
Comment (dav3r) [inline src/lambda_handler.py]: Fix copy/paste error:
```suggestion
            errors.append('"expiration_days" must be an integer.')
```

File: src/lambda_handler.py [hunk 3]
```diff
@@ -36,50 +40,159 @@ def task_default(event):
...
+        errors.append('Missing required key "expiration_days" in event.')
+    elif not event["expiration_days"]:
+        errors.append('"account_ids" must be non-null.')
     else:
         try:
-            variable_error_msg = "numerator: %s, denominator: %s"
-            result["result"] = example_div(int(numerator), int(denominator))
+            tmp = int(event["expiration_days"])
+            if tmp <= 0:
+                errors.append('"account_ids" must be a strictly positive integer.')
```
Comment (dav3r) [inline src/lambda_handler.py]: Fix copy/paste error:
```suggestion
                errors.append('"expiration_days" must be a strictly positive integer.')
```

**Review by dav3r — APPROVED** [review]
Looks great!  I noted a few thangs in `validate_event_data` that I think you overlooked.

**Review by mcdonnnj — CHANGES_REQUESTED** [review]
Please enable the dependabot ignore directives and fix the upstream repository reference in https://github.com/cisagov/disable-inactive-iam-users-lambda/blob/15fc27e20945f2a7a7945535c4a54647a6c1ce1f/.github/dependabot.yml#L22-L24

That reference changing suggests you have not updated your locally installed `skeleton` extension. Please update with `gh extension upgrade --all` to ensure you're working with the latest copy.

**Comment by jsf9k** [comment]
> Please enable the dependabot ignore directives and fix the upstream repository reference in
> 
> https://github.com/cisagov/disable-inactive-iam-users-lambda/blob/15fc27e20945f2a7a7945535c4a54647a6c1ce1f/.github/dependabot.yml#L22-L24
> 
> That reference changing suggests you have not updated your locally installed `skeleton` extension. Please update with `gh extension upgrade --all` to ensure you're working with the latest copy.

Please see commit 469a51f.

**Comment by dv4harr10** [comment]
Hi Team, for README.md file section 'How to update Python dependencies' sentence 'More information about the Pipfile format can be found here.' The link from 'here' returns a 404 not found error.

**Comment by jsf9k** [comment]
> Hi Team @ALL , for README.md file section 'How to update Python dependencies' sentence 'More information about the Pipfile format can be found here.' The link from 'here' returns a 404 not found error.

I created cisagov/skeleton-aws-lambda-python#32 to fix this upstream.

**Review by dv4harr10 — APPROVED** [review]

**Comment by jsf9k** [comment]
@mcdonnnj - This work is due tomorrow, and you are out of the office, so I am going to assume that you are satisfied with the changes I made in response to your review.

**Pull request merged**